front/app/containers/Admin/projects/project/analysis/Heatmap/SummarizeButton.tsx

```diff
@@ -69,14 +69,30 @@ const SummarizeButton = ({ row, column, cell, buttonStyle }: Props) => {
     const isUserField = userCustomFieldsIds.includes(fieldId);
     const keyPrefix = isUserField ? 'author_custom_' : 'input_custom_';
     const binType = bin.attributes.type;
+    const isAgeBin = binType === 'CustomFieldBins::AgeBin';
+    const isRangeBin = binType === 'CustomFieldBins::RangeBin';
 
-    const isRangeType = [
-      'CustomFieldBins::AgeBin',
-      'CustomFieldBins::RangeBin',
-    ].includes(binType);
     const isOptionBin = binType === 'CustomFieldBins::OptionBin';
+    if (isAgeBin) {
+      const fromKey = `${keyPrefix}${fieldId}_from`;
+      const toKey = `${keyPrefix}${fieldId}_to`;
+      // Get current year
+      const currentYear = new Date().getFullYear();
+
+      // Calculate ages from birth years
+      const ageFrom = bin.attributes.range?.end
+        ? (currentYear - bin.attributes.range.end).toString()
+        : undefined;
+      const ageTo = bin.attributes.range?.begin
+        ? (currentYear - bin.attributes.range.begin).toString()
+        : undefined;
```

Comment:
In the BE, we're offsetting this year with 6 months, since we don't know the exact birthday, it's more correct to assume people have their birthday on July 1st.

https://github.com/CitizenLabDotCo/citizenlab/blob/5a6d0e0b3ba65b26ef6c6a95271c35303b4e1573/back/engines/commercial/user_custom_fields/app/services/user_custom_fields/age_counter.rb#L28

Supporting bin filters would make this a lot easier and more centralized.